js/src/components/contact-information/store-address-card.js

```diff
@@ -92,27 +82,83 @@ export default function StoreAddressCard( { isPreview } ) {
 		<AccountCard
 			appearance={ APPEARANCE.ADDRESS }
 			description={ description }
-			hideIcon={ isPreview }
 			indicator={ editButton }
 		>
-			{ ! isPreview && (
+			<CardDivider />
+			<Section.Card.Body>
+				<Subsection.Title>
+					{ __( 'Store address', 'google-listings-and-ads' ) }
+					<AppButton
+						className="gla-store-address__refresh-button"
+						icon={ GridiconRefresh }
+						iconSize={ 16 }
+						onClick={ refetch }
+						disabled={ ! loaded }
+					/>
+				</Subsection.Title>
+				{ addressContent }
+			</Section.Card.Body>
+		</AccountCard>
+	);
+}
+
+/**
+ * Renders a component with the store address.
+ * In preview mode, meaning there will be no refresh button, just the edit link.
+ *
+ * @fires gla_edit_mc_store_address Whenever "Edit" is clicked.
+ *
+ * @param {Object} props React props
+ * @param {string} props.editHref URL where Edit button should point to.
+ * @param {JSX.Element} props.learnMore Link to be shown at the end of missing data message.
+ * @return {JSX.Element} Filled AccountCard component.
+ */
+export function StoreAddressCardPreview( { editHref, learnMore } ) {
+	const { loaded, data } = useStoreAddress( 'mc' );
+	let content, warning;
+
+	if ( loaded ) {
+		const {
+			isAddressFilled,
+			isMCAddressDifferent,
+			address,
+			address2,
+			city,
+			state,
+			country,
+			postcode,
+		} = data;
+		const stateAndCountry = state ? `${ state } - ${ country }` : country;
+
+		if ( isAddressFilled && ! isMCAddressDifferent ) {
+			content = [ address, address2, city, stateAndCountry, postcode ]
+				.filter( Boolean )
+				.join( ', ' );
+		} else {
+			warning = __(
+				'Please add your store address',
+				'google-listings-and-ads'
+			);
+			content = (
 				<>
-					<CardDivider />
-					<Section.Card.Body>
-						<Subsection.Title>
-							{ __( 'Store address', 'google-listings-and-ads' ) }
-							<AppButton
-								className="gla-store-address__refresh-button"
-								icon={ GridiconRefresh }
-								iconSize={ 16 }
-								onClick={ refetch }
-								disabled={ ! loaded }
-							/>
-						</Subsection.Title>
-						{ addressContent }
-					</Section.Card.Body>
+					{ __(
+						'Google requires the store address for all stores using Google Merchant Center. ',
+						'google-listings-and-ads'
+					) }
+					{ learnMore }
 				</>
-			) }
-		</AccountCard>
+			);
+		}
+	}
+
+	return (
+		<ContactInformationPreviewCard
+			appearance={ APPEARANCE.ADDRESS }
+			editHref={ editHref }
+			editEventName="gla_edit_mc_store_address"
```

Comment:
❓ 
The `gla_edit_mc_store_address` event in [the `<StoreAddressCard>`](https://github.com/woocommerce/google-listings-and-ads/blob/09f25d86dab2b66e2d86d5bb2795650fff777ba4/js/src/components/contact-information/store-address-card.js#L49) is original triggered when opening the WC Settings page to a new tab. But the same event here is triggered by [the `<ContactInformationPreviewCard>`](https://github.com/woocommerce/google-listings-and-ads/blob/09f25d86dab2b66e2d86d5bb2795650fff777ba4/js/src/components/contact-information/contact-information-preview-card.js#L44) when navigating to the edit address page. Seems they shouldn't be aggregated to the same event?

Reply:
My initial thought was:

The original description of this event was
> `edit_mc_store_address` - Trigger when store address edit button is clicked.

So both cases seem to match, and `path`+`subpath` could differentiate those cases.

If we would like to have split those events, I'd go for 
- `edit_mc_store_address` - Trigger when store address "Edit" button is clicked. For <code>&lt;StoreAddressCard<b>Preview</b>&gt;</code> usage. As it opens a GLA page to edit **MC** store address
- `edit_wc_store_address` - Trigger when store address "Edit in settings" button is clicked. For `<StoreAddressCard>` usage (on `/wp-admin/admin.php?page=wc-admin&subpath=%2Fedit-store-address&path=%2Fgoogle%2Fsettings`). As it opens a page to edit only **WC** store address.

Reply:
Updated in https://github.com/woocommerce/google-listings-and-ads/pull/1025/commits/54aea65d3f1c6a35758d9fdaa9a64ac676156970

Reply:
I also drafted the event registration PR: 588-gh-Automattic/tracks-events-registration